packages/jest-snapshot/src/utils.js

```diff
@@ -66,6 +66,12 @@ const serialize = (data: any): string => {
   }));
 };
 
+const unescape = (data: any): string => {
+  return data
+    .replace(/\\(")/g, '$1') // unescape double quotes
+    .replace(/\\([\\^$*+?.()|[\]{}])/g, '$1'); // then unescape RegExp
```

Comment:
We could avoid unescaping regex, it will result in many backslashes, but it's still gonna be usable.